### PR TITLE
print compose logs to console in dev mode, add license mount, sync conf

### PR DESCRIPTION
**Pull request by riuvshin**
1) in dev mode we have to print compose output to console to make devs life easier
2) added missing mount for license file to not loss it after restart
3) sync script with deployment

@garagatyi @TylerJewell @benoitf

### CODE REVIEW

**Comment by TylerJewell** [comment]
+1 - this is a great idea.

File: dockerfiles/cli/cmd_start.sh [hunk 1]
```diff
@@ -53,8 +53,13 @@ cmd_start() {
...
-  docker_compose --file="${REFERENCE_CONTAINER_COMPOSE_FILE}" \
-                 -p=$CHE_MINI_PRODUCT_NAME up -d >> "${LOGS}" 2>&1
+  COMPOSE_UP_COMMAND="docker_compose --file=\"${REFERENCE_CONTAINER_COMPOSE_FILE}\" -p=\"${CHE_MINI_PRODUCT_NAME}\" up -d"
+
+  if [ "${CODENVY_DEVELOPMENT_MODE}" != "on" ]; then
+    COMPOSE_UP_COMMAND+=" >> \"${LOGS}\" 2>&1"
+  fi
+
+  log ${COMPOSE_UP_COMMAND}
+  eval ${COMPOSE_UP_COMMAND}
```
Comment (benoitf) [inline dockerfiles/cli/cmd_start.sh]: should we use at some point cli_eval/silent_eval functions instead of eval with some flags

I'm not sure in dev mode we need to see this log everytime on the console. Looks like each developer has its view on this, so I would prefer a toggle flag

File: modules/codenvy/templates/cleanUpWorkspaceStorage.sh.erb [hunk 2]
```diff
@@ -1,6 +1,16 @@
 #!/bin/sh
 
 DST_FOLDER=${1}
-CLEANUP_REGEX='^(/opt/codenvy-data/fs/([0-9a-f]{2}/){3}workspace[0-9a-z]+)$'
```
Comment (garagatyi) [inline modules/codenvy/templates/cleanUpWorkspaceStorage.sh.erb]: Why did you change this file?

File: modules/compose/templates/docker-compose-container.yml.erb [hunk 3]
```diff
@@ -146,6 +146,7 @@ services:
        - '<%= scope.lookupvar('compose::codenvy_folder') -%>/logs/codenvy/:/opt/codenvy-data/logs/'
        - '<%= scope.lookupvar('compose::codenvy_folder') -%>/logs/codenvy/che-machines-logs/:/opt/codenvy-data/che-machines-logs/'
        - '<%= scope.lookupvar('compose::codenvy_folder') -%>/data/codenvy/fs/:/opt/codenvy-data/fs/'
+       - '<%= scope.lookupvar('compose::codenvy_folder') -%>/data/codenvy/license/:/opt/codenvy-data/license/'
```
Comment (garagatyi) [inline modules/compose/templates/docker-compose-container.yml.erb]: Why did you change this file?

File: modules/codenvy/templates/cleanUpWorkspaceStorage.sh.erb (hunk 2 above)
Comment (riuvshin) [inline modules/codenvy/templates/cleanUpWorkspaceStorage.sh.erb]: bo bikoz

File: modules/compose/templates/docker-compose-container.yml.erb (hunk 3 above)
Comment (riuvshin) [inline modules/compose/templates/docker-compose-container.yml.erb]: without mount /license if license is generated it will be gone after codenvy restart because it is not mounter

File: dockerfiles/cli/cmd_start.sh (hunk 1 above)
Comment (riuvshin) [inline dockerfiles/cli/cmd_start.sh]: maybe we can improve it like this, my only motivation here is that due to frequent changes in CLI guys fail to start codenvy and they come to me and saying it didn't work. if they will see error message on console it will be much easier for them to figure out where is a problem.

File: modules/codenvy/templates/cleanUpWorkspaceStorage.sh.erb [hunk 4]
```diff
@@ -1,6 +1,16 @@
...
+CLEANUP_REGEX='^(.*/codenvy-data/fs/([0-9a-f]{2}/){3}workspace[0-9a-z]+)$'
 
-[[ ${DST_FOLDER} =~ ${CLEANUP_REGEX} ]] && sudo rm -rf "${DST_FOLDER}"
+hasSudoCommand() {
+  hash sudo 2>/dev/null && return 0 || return 1
+}
+
+if echo "${DST_FOLDER}" | grep -q -E "${CLEANUP_REGEX}"; then
```
Comment (riuvshin) [inline modules/codenvy/templates/cleanUpWorkspaceStorage.sh.erb]: that script written in a way to be compatible with old codenvy where we have sudo. So for now we want to keep it like that and once codenvy will be only in docker we will clean this up.

File: dockerfiles/cli/cmd_start.sh (hunk 1 above)
Comment (benoitf) [inline dockerfiles/cli/cmd_start.sh]: I think it would great to have it displayed on console if there are errors

Comment (riuvshin) [inline dockerfiles/cli/cmd_start.sh]: +1 but I don't see now how to do that, I think it is more about refactoring in a general way for managing that
